apps/crn-server/test/data-providers/contentful/calendars.data-provider.test.ts

```diff
@@ -288,6 +397,74 @@ describe('Calendars data provider', () => {
       expect(calendarMock.update).toHaveBeenCalled();
       expect(calendarMockUpdated.publish).toHaveBeenCalled();
     });
+
+    test('Should update the calendar - with new google api metadata - without previous google api metadata', async () => {
+      const calendarId = 'calendar-id-1';
+
+      const calendarMock = getEntry({});
+      environmentMock.getEntry.mockResolvedValueOnce(calendarMock);
+      const calendarMockUpdated = getEntry({});
+      calendarMock.update = jest
+        .fn()
+        .mockResolvedValueOnce(calendarMockUpdated);
+
+      await calendarDataProviderMock.update(calendarId, {
+        syncToken: 'token-1',
+      });
+
+      expect(environmentMock.getEntry).toHaveBeenCalledWith(calendarId);
+
+      expect(calendarMock.fields).toEqual({
+        googleApiMetadata: {
+          'en-US': {
+            syncToken: 'token-1',
+          },
+        },
+      });
+      expect(calendarMock.update).toHaveBeenCalled();
+      expect(calendarMockUpdated.publish).toHaveBeenCalled();
+    });
+
+    test('Should update the calendar - with previous google api metadata', async () => {
+      const calendarId = 'calendar-id-1';
+
+      const calendarMock = getEntry({
+        googleCalendarId: {
+          'en-US': 'google-calendar-id-1',
+        },
+        googleApiMetadata: {
+          'en-US': {
+            syncToken: 'syncToken-1',
+          },
+        },
+      });
+      environmentMock.getEntry.mockResolvedValueOnce(calendarMock);
+      const calendarMockUpdated = getEntry({});
+      calendarMock.update = jest
+        .fn()
+        .mockResolvedValueOnce(calendarMockUpdated);
+
+      await calendarDataProviderMock.update(calendarId, {
+        resourceId: 'resourceId-1',
+      });
+
+      expect(environmentMock.getEntry).toHaveBeenCalledWith(calendarId);
+
+      expect(calendarMock.fields).toEqual({
+        googleCalendarId: {
+          'en-US': 'google-calendar-id-1',
+        },
+        googleApiMetadata: {
+          'en-US': {
+            associatedGoogleCalendarId: 'google-calendar-id-1',
```

Comment:
It updates the `associatedGoogleCalendarId` when we update the `resourceId`. I've left a comment in the data provider (https://github.com/yldio/asap-hub/blob/CT-9-google-calendar-sync-with-json-metadata/apps/crn-server/src/data-providers/contentful/calendars.data-provider.ts#L135:L136). 

The sync token is updated later in the eventsUpdated handler. I don't think we need to update `associatedGoogleCalendarId` again since it should be the same as the one when we subscribed to the calendar. Do you see the need to update the `associatedGoogleCalendarId` when the `syncToken` gets updated? And if so, why? :)